tests/cases/fourslash/completionsImport_reExportDefault2.ts

```diff
@@ -0,0 +1,39 @@
+/// <reference path="fourslash.ts" />
+
+// @module: preserve
+// @checkJs: true
+
+// @Filename: /node_modules/example/package.json
+//// { "name": "example", "version": "1.0.0", "main": "dist/index.js" }
+
+// @Filename: /node_modules/example/dist/nested/module.d.ts
+//// declare const defaultExport: () => void;
+//// declare const namedExport: () => void;
+//// 
+//// export default defaultExport;
+//// export { namedExport };
+
+// @Filename: /node_modules/example/dist/index.d.ts
+//// export { default, namedExport } from "./nested/module";
```

Comment:
The ExportInfoMap stores chains of re-exports of a given symbol for as long as the symbol has “the same name” in every link in the chain. And by “name,” we mean ones that the user would want to reference a symbol by, not `default` or `export=`. So when we see `default` or `export=`, we try to figure out a better name to offer as the auto-import, by trying to resolve the default export to another named declaration, or making a name out of the filename as a last resort. We do this (among other places) as we’re building the chains of re-exports for the ExportInfoMap, with the goal of being able to answer the question “Give me all the modules from which I can import a given symbol by a given name.” For this test to work, we need to understand that we can import `defaultExport` declared in `example/dist/nested/module.d.ts` both from there and from this re-export right here ☝️. But before this PR, when processing _this_ re-exported `default` symbol, we didn’t search its target file for local names. Instead, we just decided to call it `dist`, after `dist/index.d.ts`, and so it created a new chain in the map instead of being joined to `defaultExport` from `example/dist/nested/module.d.ts`.